add README in advance of archiving (#452)
golang-fips/openssl-fips: README.md

# Go OpenSSL bindings for FIPS compliance

> [!IMPORTANT]

> > **This repository has been archived.** Active development continues at [github.com/microsoft/go-crypto-openssl](https://github.com/microsoft/go-crypto-openssl).

Reviewer: ```suggestion
> **This repository is no longer active.** Active development continues at [github.com/microsoft/go-crypto-openssl](https://github.com/microsoft/go-crypto-openssl).
```

[![Go Reference](https://pkg.go.dev/badge/github.com/golang-fips/openssl/v2.svg)](https://pkg.go.dev/github.com/golang-fips/openssl/v2)

The `openssl` package implements Go crypto primitives using OpenSSL shared libraries and cgo. When configured correctly, OpenSSL can be executed in FIPS mode, making the `openssl` package FIPS compliant.

The `openssl` package is designed to be used as a drop-in replacement for the [boring](https://pkg.go.dev/crypto/internal/boring) package in order to facilitate integrating `openssl` inside a forked Go toolchain.

## A Note on This Archive

> Red Hat and Microsoft have decided to take a different strategy for FIPS-compliant cryptography in Go going forward. As a result, this repository is now archived, and active development continues at [microsoft/go-crypto-openssl](https://github.com/microsoft/go-crypto-openssl).

Reviewer: A bit clearer, and with a link to more details:

```suggestion
[As of May 2026, upstream Go cryptography has a CMVP certificate, and Red Hat has decided to use upstream cryptography rather than continue to integrate this module in a Go fork](https://github.com/golang-fips/go#migration-to-upstream-fips-certified-cryptography). However, Microsoft will continue to use OpenSSL as a cryptography backend for Go FIPS compliance. As a result, active development continues at [microsoft/go-crypto-openssl](https://github.com/microsoft/go-crypto-openssl).
```
